src/scp/NominationProtocol.cpp

```diff
@@ -254,6 +254,15 @@ NominationProtocol::updateRoundLeaders()
             }
             return true;
         });
+
+        if (topPriority == 0)
+        {
+            // No one had priority, so all nodes would choose themselves
+            // resulting in a timeout. Clear newRoundLeaders, allowing the
+            // algorithm to fast timeout and try again.
+            newRoundLeaders.clear();
+        }
```

Comment:
I want to call this out again to double check my thinking about the worst-case scenario with this change. If some nodes are running the new algorithm and some are running the old algorithm (because they have `QUORUM_SET` manually defined), then it's possible that the nodes running the new algorithm hit this case and fast timeout, while the nodes running the old algorithm do not fast timeout. In that case, their `mRoundNumber` values diverge. This has an impact in a few places:

1. timeout lengths differ. I don't think this is a big problem as it's equivalent to some nodes having a worse connection than others.
2.  `hashNode` values differ. `hashNode` is used for the priority function, which already differs between the two algorithms. We've already reasoned about the impact of nodes running each algorithm choosing different leaders (potential timeouts) and are OK with them.
3. `hashValue` values differ. The impact of this is that nodes running each algorithm may vote for different values initially. However, over time nodes expand their sets of voted for values and they should still eventually converge on some values with enough votes to cross the threshold. The impact here is potential additional timeouts while waiting for convergence.

Simulation has demonstrated that this `topPriority == 0` case is rare when modeling Tier 1 (affecting ~0.04% of slots). It will become even rarer as we expand the validator set. Nevertheless, part of rolling this change out is communication with validators to ensure they're using automatic quorum set configuration, and therefore will end up using the new algorithm after the vote.

Does this reasoning sound correct? @MonsieurNicolas @marta-lokhova

Reply:
yeah I agree that round numbers diverging is not the worst problem as rounds are supposed to be "additive" (as in: if a node is behind, it may actually unstick a node that is ahead as the ones ahead are supposed to listen to the union of leaders from previous rounds).

For the divergence in hashing, maybe we should have a flag to force using the old behavior (based on the generated quorum set) for situations where this would be a problem?

Reply:
nm -- the flag is just `FORCE_OLD_STYLE_LEADER_ELECTION` so I think we're good